File: backend/foodNwaste/foodNwaste/src/main/java/com/capstone/foodNwaste/model/FoodNWaste.java
package com.capstone.foodNwaste.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class FoodNWaste {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id; // Primary key

    private Long userId;

    private LocalDate date;

    private double amountSpent;

    private String foodType; // "Veg" or "NonVeg"

    private double waterUsage; // Water usage in liters
}

File: FrontEnd/src/app/models/FoodNWaste.ts
export class FoodWaste {
    userId: number;
    date: string;
    amountSpent: number;
    foodType: string;
    waterUsage: number;
  
    constructor(
      userId: number,
      date: string,
      amountSpent: number,
      foodType: string,
      waterUsage: number
    ) {
      this.userId = userId;
      this.date = date;
      this.amountSpent = amountSpent;
      this.foodType = foodType;
      this.waterUsage = waterUsage;
    }
  }
  